delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,825 @@
+                    if not query_para or \
+                            (int(query_para['begin_time'])
+                             <= occur_time
+                             <= int(query_para['end_time'])):
+                        description = ''
+                        for i in range(4, len(value_array)):
+                            description += value_array[i] + ' '
+                        severity = constant.SEVERITY_MAP.get(value_array[1])
+                        alert_model = {
+                            'alert_id': value_array[0],
+                            'alert_name': value_array[0],
+                            'severity': severity,
+                            'category': constants.Category.FAULT,
+                            'type': constants.EventType.EQUIPMENT_ALARM,
+                            'occur_time': occur_time,
+                            'description': description,
+                            'match_key': hashlib.md5(
+                                (value_array[0] + severity +
+                                 description).encode()).hexdigest(),
+                            'resource_type': constants.DEFAULT_RESOURCE_TYPE,
+                            'location': ''
+                        }
+                        alert_list.append(alert_model)
+            return alert_list
+        except exception.DelfinException as e:
+            err_msg = "Failed to get alerts from " \
+                      "hitachi nas: %s" % (six.text_type(e.msg))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get alerts from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    @staticmethod
+    def parse_alert(alert):
+        try:
+            alert_info = alert.get(constant.OID_TRAP_DATA)
+            alert_array = alert_info.split(":")
+            if len(alert_array) > 1:
+                description = alert_array[1]
+                alert = alert_array[0].split()
+                if len(alert) > 1:
+                    alert_id = alert[0]
+                    severity = constant.SEVERITY_MAP.get(alert[1])
+                    alert_model = {
+                        'alert_id': alert_id,
+                        'alert_name': alert_id,
+                        'severity': severity,
+                        'category': constants.Category.FAULT,
+                        'type': constants.EventType.EQUIPMENT_ALARM,
+                        'occur_time': utils.utcnow_ms(),
+                        'description': description,
+                        'match_key': hashlib.md5(
+                            (alert_id + severity +
+                             description).encode()).hexdigest(),
+                        'resource_type': constants.DEFAULT_RESOURCE_TYPE,
+                        'location': ''
```

Comment:
If info cannot be attained from storage, it can be deleted.

Reply:
has been modify